DependencyInjection/HWIOAuthExtension.php

```diff
@@ -43,16 +48,41 @@ public function load(array $configs, ContainerBuilder $container)
         $config = $processor->processConfiguration(new Configuration(), $configs);
 
         // setup http client settings
-        $httpClient = $container->getDefinition('hwi_oauth.http_client');
-        $httpClient->addMethodCall('setVerifyPeer', array($config['http_client']['verify_peer']));
-        $httpClient->addMethodCall('setTimeout', array($config['http_client']['timeout']));
-        $httpClient->addMethodCall('setMaxRedirects', array($config['http_client']['max_redirects']));
-        $httpClient->addMethodCall('setIgnoreErrors', array($config['http_client']['ignore_errors']));
-        $httpClient->addMethodCall('setOption', array(CURLOPT_ENCODING, ''));
-        if (isset($config['http_client']['proxy']) && $config['http_client']['proxy'] != '') {
-            $httpClient->addMethodCall('setProxy', array($config['http_client']['proxy']));
+        $guzzleConfig = array(
```

Comment:
Perhaps we can integrate `php-http/httplug-bundle` to handle configuration stuff?